src/NServiceBus.Core/Routing/UnicastRouter.cs

```diff
@@ -22,16 +23,20 @@ public UnicastRouter(MessageMetadataRegistry messageMetadataRegistry, UnicastRou
             this.physicalAddresses = physicalAddresses;
         }
 
-        public IEnumerable<UnicastRoutingStrategy> Route(Type messageType, DistributionStrategy distributionStrategy, ContextBag contextBag)
+        public async Task<IEnumerable<UnicastRoutingStrategy>> Route(Type messageType, DistributionStrategy distributionStrategy, ContextBag contextBag)
         {
             var typesToRoute = messageMetadataRegistry.GetMessageMetadata(messageType)
                 .MessageHierarchy
                 .Distinct()
                 .ToList();
 
-            var routes = typesToRoute.SelectMany(t => unicastRoutingTable.GetDestinationsFor(t, contextBag)).Distinct().ToList();
+            var routes = new List<IUnicastRoute>();
+            foreach (var routeType in typesToRoute)
+            {
+                routes.AddRange(await unicastRoutingTable.GetDestinationsFor(routeType, contextBag).ConfigureAwait(false));
```

Comment:
Yes, should be possible for this. It can be pushed to the DB as a single operation.